runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java

```diff
@@ -100,83 +100,88 @@ public ChatMessageManager cacheColor(final ChatColor chatColor, final ChatMessag
+		final ChatLineBuffer chatLineBuffer = client.getChatLineMap().get(message.getType().getType());
 		final MessageNode[] lines = chatLineBuffer.getLines();
 		final MessageNode line = lines[0];
-		update(line.getType(), mesage, line);
+
+		// Update the message with RuneLite additions
+		line.setRuneLiteFormatMessage(message.getRuneLiteFormattedMessage());
+		update(line);
 	}
 
-	public void update(final ChatMessageType type, final String message, final MessageNode target)
+	public void update(final MessageNode target)
 	{
+		if (Strings.isNullOrEmpty(target.getRuneLiteFormatMessage()))
+		{
+			return;
+		}
+
 		final Client client = clientProvider.get();
-		final Set<ChatColor> chatColors = colorCache.get(type);
+		final boolean transparent = client.isResized() && client.getSetting(Varbits.TRANSPARANT_CHATBOX) != 0;
+		final Set<ChatColor> chatColors = colorCache.get(target.getType());
 
-		// If we do not have any colors cached or recoloring is disabled, simply set message
+		// If we do not have any colors cached or recoloring is disabled, simply set clean message
 		if (!config.chatCommandsRecolorEnabled() || chatColors == null || chatColors.isEmpty())
 		{
-			target.setRuneLiteFormatMessage(message);
-			target.setValue(message);
+			target.setValue(target.getRuneLiteFormatMessage());
 			return;
 		}
 
+		target.setValue(recolorMessage(transparent, target.getRuneLiteFormatMessage(), target.getType()));
+	}
+
+	private String recolorMessage(boolean transparent, String message, ChatMessageType messageType)
+	{
+		final Set<ChatColor> chatColors = colorCache.get(messageType);
 		final AtomicReference<String> resultMessage = new AtomicReference<>(message);
 
 		// Replace custom formatting with actual colors
 		chatColors.stream()
-			.filter(chatColor -> chatColor.isTransparent() ==
-				(client.isResized() && client.getSetting(Varbits.TRANSPARANT_CHATBOX) != 0))
+			.filter(chatColor -> chatColor.isTransparent() == transparent)
 			.forEach(chatColor ->
 				resultMessage.getAndUpdate(oldMessage -> oldMessage.replaceAll(
 					"<col" + chatColor.getType().name() + ">",
 					"<col=" + Integer.toHexString(chatColor.getColor().getRGB() & 0xFFFFFF) + ">")));
 
-		target.setRuneLiteFormatMessage(message);
-		target.setValue(resultMessage.get());
+		return resultMessage.get();
 	}
 
 	public void refreshAll()
 	{
-		if (!config.chatCommandsRecolorEnabled())
-		{
-			return;
-		}
-
 		final Client client = clientProvider.get();
 
 		executor.submit(() ->
 		{
-			final Set<MessageNode> chatLines = client.getChatLineMap().values().stream()
+			client.getChatLineMap().values().stream()
 				.filter(Objects::nonNull)
 				.flatMap(clb -> Arrays.stream(clb.getLines()))
 				.filter(Objects::nonNull)
-				.filter(mn -> mn.getRuneLiteFormatMessage() != null)
-				.collect(Collectors.toSet());
-
-			chatLines.forEach(chatLine -> update(chatLine.getType(), chatLine.getRuneLiteFormatMessage(), chatLine));
+				.forEach(this::update);
```

Comment:
Probably how I should've done this initially here is to use reduce and return boolean from update, and in case the reduce returns true, call the refresh-chat.